scripts/core/ui/components/ListPage/generic-list-page.tsx

```diff
@@ -613,12 +601,33 @@ export class GenericListPageComponent<T extends IBaseRestApiResponse>
     }
 }
 
-export const getGenericListPageComponent = <T extends IBaseRestApiResponse>(resource: string) =>
-    connectServices<IPropsGenericForm<T>>(
-        connectCrudManager<IPropsGenericForm<T>, IPropsConnected<T>, T>(
-            GenericListPageComponent,
-            'items',
-            resource,
-        )
-        , ['modal', '$rootScope', 'notify'],
-    );
+export const getGenericListPageComponent =
+    <T extends IBaseRestApiResponse>(resource: string, formConfig?: IFormGroup) =>
```

Comment:
Can you make `formConfig` non-optional and pass it from all the places that call the function?

Reply:
@tomaskikutis what about [here](https://github.com/superdesk/superdesk-client-core/blob/master/scripts/extensions/annotationsLibrary/src/AnnotationInputWithKnowledgeBase.tsx#L49), as you can see it's not using any formConfig but it's [generating the filters](https://github.com/superdesk/superdesk-client-core/blob/master/scripts/extensions/annotationsLibrary/src/AnnotationInputWithKnowledgeBase.tsx#L27) anyway

Reply:
Do you see an issue with it? `formatConfig` is needed for retrieving the field type and passing it along with a value to `generateFilterForServer`. Here it's done directly.

Reply:
it's not an issue if `formConfig` is optional in `connectCrudManager`

Reply:
ok nevermind I thought you were saying formConfig should be non-optional in crudmanager, not genericListPageComponent